Service/ReplicaSettingsHandler.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service;
+
+use Algolia\AlgoliaSearch\Api\Data\IndexOptionsInterface;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ *  This class will proxy the settings save operations and forward to replicas based on user configuration
+ *  excluding attributes which should never be forwarded
+ */
+class ReplicaSettingsHandler
```

Comment:
Since this class is responsible for applying settings on primary AND replicas, I would assume it to be called SettingsHandler (not blocking though, I'm just picky)

Reply:
Yeah I struggled with naming this because SettingsHandler seemed too generic and I wanted to communicate the business use case in that it is specifically needed to handle replicas. What if I called it `ReplicaAwareSettingsHandler`? Do you like that better or too verbose?

Reply:
Yeah, I'm not a huge fan to add more and more words to the name 😅 

What about `IndexSettingsHandlers` ?